dds/src/dcps/infrastructure/instance.rs

```diff
@@ -1,36 +1,123 @@
-use core::ops::Index;
+use crate::transport::types::Guid;
+use core::{
+    borrow::Borrow,
+    ops::{Deref, Index},
+};
 use dust_dds_derive::TypeSupport;
 
-/// Type for the instance handle representing an Entity
+/// Special constant value representing a 'nil' [`InstanceHandle`].
+pub const HANDLE_NIL: InstanceHandle = InstanceHandle([0; 16]);
+
+/// Type for the instance handle representing an Entity.
+#[repr(transparent)]
 #[derive(Clone, Copy, PartialEq, Eq, Debug, Hash, PartialOrd, Ord, TypeSupport)]
 pub struct InstanceHandle([u8; 16]);
 
 impl InstanceHandle {
-    /// InstanceHandle constructor
+    /// Constructs a new `InstanceHandle`.
+    #[inline]
     pub const fn new(bytes: [u8; 16]) -> Self {
-        InstanceHandle(bytes)
+        Self(bytes)
+    }
+}
+
+impl From<[u8; 16]> for InstanceHandle {
+    #[inline]
+    fn from(value: [u8; 16]) -> Self {
+        Self(value)
+    }
+}
+
+impl From<InstanceHandle> for [u8; 16] {
+    #[inline]
+    fn from(value: InstanceHandle) -> Self {
+        value.0
+    }
+}
+
+impl From<Guid> for InstanceHandle {
+    #[inline]
+    fn from(value: Guid) -> Self {
+        Self(value.into())
+    }
+}
+
+impl From<InstanceHandle> for Guid {

```

Comment:
A Guid can be converted into an InstanceHandle but an InstanceHandle is not guaranteed to be convertible into a Guid so I would suggest to delete this trait.

Reply:
I need to extract `GuidPrefix` (`[u8; 12]`) from an `InstanceHandle`: `InstanceHandle` -> `Guid` -> `GuidPrefix`.
Why is this approach incorrect? If it is wrong, what's the correct way to do this?